File: quickSort.ts
/**
 * Quick sort implementation.
 * @param data - List to be ordered.
 * @param reverse - If true, sort in descending order; otherwise, sort in ascending order.
 * @returns Ordered list.
 */
function quickSort(data: number[], reverse: boolean = false): number[] {
  if (data.length <= 1) {
    return data;
  }

  const pivot: number = data[Math.floor(data.length / 2)];
  const left: number[] = data.filter((x) => x < pivot);
  const middle: number[] = data.filter((x) => x === pivot);
  const right: number[] = data.filter((x) => x > pivot);

  const soerted_array: number[] = quickSort(left).concat(middle, quickSort(right));

  if (reverse) {
    const n: number = data.length;
    const newArray: number[] = [];
    // this "for" is used to avoid the use of the array.reverse() built in function
    for (let i = 0; i < n; i++) {
      soerted_array[i] = data[n - i];
    }
    return newArray;
  }
  return soerted_array;

}

File: quickSort.py
"""Quick sort implementation."""


def quick_sort(data: list, reverse: bool = False):
    """Quick sorting function.
    Args:
        data (list): list to be ordered.
        reverse (bool, optional): ascending(False) or descending(True) order. Defaults to False.
    Returns:
        list: ordered list.
    """
    if len(data) <= 1:
        return data
    pivot = data[len(data) // 2]
    left = [x for x in data if x < pivot]
    middle = [x for x in data if x == pivot]
    right = [x for x in data if x > pivot]

    if reverse:
        return quick_sort(right)[::-1] + middle[::-1] + quick_sort(left)[::-1]

    return quick_sort(left) + middle + quick_sort(right)
